core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -198,6 +198,14 @@ protected void cleanUncommitted(Set<ManifestFile> committed) {
     }
   }
 
+  @Override
+  protected boolean cleanUncommittedAfterCommit() {
+    // appendManifests are not rewritten, never need cleanup
+    // rewrittenAppendManifests are rewritten in appendManifest, never need cleanup
+    // newManifests are cleaned up in writeNewManifests
```

Comment:
Nit: I'd make this a method level comment instead of inlining all this.

```
/**
Cleanup after committing is disabled for FastAppend for the following reasons:

1.) Appended manifests are never rewritten
2.) Manifests which are written out as part of appendFile are cleaned up between commit attempts 
in writeNewManifests
*/
```

Reply:
> rewrittenAppendManifests are rewritten in appendManifest, never need cleanup

Actually, I'll need to take a double pass on the `rewrittenAppendManifests` case, I'm not 100% sure yet this is the case. In the worst case though if we miss an opportunity to cleanup, orphan file removal would always pick it up anyways.

Reply:
You are correct, changed the condition to `rewrittenAppendManifests.size() > 1`.